src/mo_frontend/typing.ml

```diff
@@ -3068,6 +3108,7 @@ and infer_obj env obj_sort exp_opt dec_fields at : T.typ =
   leave_scope env (private_identifiers scope.Scope.val_env) initial_usage;
   let (_, tfs) = T.as_obj t in
   if not env.pre then begin
+    (* TODO: Mixins need a similar, but different check here *)
```

Comment:
Is this still to be implemented?

I think we may not have to worry about system methods as long as they enter the mixin environment - then duplicate declarations between the mix and actor will get caught.

Reply:
I can't remember, did we check the system inspect method has the correct domain (including all the mixins?) I think we did...